### ISSUE TITLE
PHP Warning:  array_intersect_key(): Argument #1 is not an array

### ISSUE THREAD

**Issue by rmcaninch** [sblaisot/automatic_addressbook#11]
Seeing this warning in the Roundcube error log.

`PHP Warning:  array_intersect_key(): Argument #1 is not an array in /usr/share/roundcube/program/steps/addressbook/save.inc on line 178`

Seems to be when the contact is edited and moved to the default address book.

**Comment by sblaisot** [comment]
Hi,
What version of roundcube are you using ? 
What version of automatic_addressbook plugin are you using ?

regards,

sb

**Comment by sblaisot** [comment]
You probably get this message when updating an existing contact in you default addressbook or something like that.
To help me investigate further, can you decribe step by step what you do to reproduce this bug / to get this message in the log.

thanks,

sb

**Comment by rmcaninch** [comment]
Using Roundcube 1.1.5

I edited an entry in the Collected Addresses, the entry did move to my default address book. Just wanted to check on that warning.

Edited to add, I don't see this warning when adding a new entry directly to the default SQL address book.

**Comment by sblaisot** [comment]
Thank for your feedback.

It's probably a bug in roundcube itself. If I understand correctly, it doen't handle very well the move of the contact entry from the collected_addressbook to the default addressbook while editing when refreshing the contact list at the end of the process.

This is probably a regression in a recent version of roundcube.

This message is mostly harmless. I opened a bug on roundcube directly for that.

Closing this one now.

sb
